packages/dds/tree/src/test/simple-tree/api/treeApi.spec.ts

```diff
@@ -768,5 +768,114 @@ describe("treeNodeApi", () => {
 			assert.equal(nodeChanged, true, "'nodeChanged' should have fired");
 			assert.equal(treeChanged, true, "'treeChanged' should have fired");
 		});
+
+		it(`'nodeChanged' includes the names of changed properties (objectNode)`, () => {
+			const sb = new SchemaFactory("test");
+			class TestNode extends sb.object("root", {
+				prop1: sb.optional(sb.number),
+				prop2: sb.optional(sb.number),
+				prop3: sb.optional(sb.number),
+			}) {}
+
+			const view = getView(new TreeViewConfiguration({ schema: TestNode }));
+			view.initialize({ prop1: 1, prop2: 2 });
+			const root = view.root;
+
+			const eventLog: ReadonlySet<string>[] = [];
+			Tree.on(root, "nodeChanged", ({ changedProperties }) =>
+				eventLog.push(changedProperties),
+			);
+
+			const { forkView, forkCheckout } = getViewForForkedBranch(view);
```

Comment:
For my knowledge: is this the recommended pattern for bundling multiple tree operations into a single transaction?

Reply:
Ah, I see this is a newly added function and not exported. Do we have a recommended pattern for external users?

Reply:
if you want to group operations, use transactions.

This code is using branches, presumably to simulate if the ops were applied by another client, but without requiring a runtime, multiple tree instances etc. This is a common pattern in our tests since branches and multiple clients hit the same code paths, but using branches involves far fewer dependencies.

This same scenario however would work fine without the branch since local edits also trigger events, so its unclear why its making a branch.